include/platform/tasklet.h

```diff
@@ -0,0 +1,88 @@
+/*
```

Comment:
Generally, any platform APIs should carry the `otPlat*()` prefix and are declared in `include/platform`.  Any APIs that are calls into the OpenThread library do not carry the `otPlat*()` prefix and are declared outside `include/platform`.  I think most of the function declarations (excluding `otSignalTaskletPending`) should live in `include/openthread-tasklet.h`.

As for what to do with `otSignalTaskletPending`, a couple options:

1. Declare a `otPlatTaskletsPending` in `include/platform/tasklet.h`.
2. Define a function pointer and API for setting up the callback in `include/platform/openthread-tasklet.h`.

Thoughts?